django/views/generic/base.py

```diff
@@ -62,9 +62,13 @@ def view(request, *args, **kwargs):
             self = cls(**initkwargs)
             if hasattr(self, 'get') and not hasattr(self, 'head'):
                 self.head = self.get
-            self.request = request
-            self.args = args
-            self.kwargs = kwargs
+            self.init(request, *args, **kwargs)
+            assert hasattr(self, 'request'), (
```

Comment:
I think `if not hasattr(...)` and raising an exception would be more in line with Django's style.